servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -3696,14 +3707,69 @@ private void populateDomainRoleMembersMapFromResultSet(Map<String, List<DomainRo
 
         // get admin roles of pending selfserve requests
 
-        getRecipientRoleForSelfServeMembershipApproval(caller, targetRoles);
+        getRecipientRoleForSelfServeMembershipApproval(caller, targetRoles, timestamp, server);
 
         return targetRoles;
     }
 
-    private void getRecipientRoleForSelfServeMembershipApproval(String caller, Set<String> targetRoles) {
+    @Override
+    public List<Map<String, String>> processExpiredPendingMembers(int pendingRoleMemberLifespan, String monitorIdentity) {
+
+        boolean result = true;
+        String caller = "processExpiredPendingMembers";
+        //update audit log with details before deleting
+        Map<String, String> membersMap;
+        String member;
+        List<Map<String, String>> membersMapList = new ArrayList<>();
+        try (PreparedStatement ps = con.prepareStatement(SQL_GET_EXPIRED_PENDING_ROLE_MEMBERS)) {
+            ps.setInt(1, pendingRoleMemberLifespan);
+            try (ResultSet rs = executeQuery(ps, caller)) {
+                while (rs.next()) {
+                    membersMap = new HashMap<>();
+                    membersMap.put("domain", rs.getString(4));
+                    membersMap.put("role", rs.getString(3));
+                    member = rs.getString(2);
+                    membersMap.put("member", member);
+                    membersMapList.add(membersMap);
+                    result = result && insertRoleAuditLog(rs.getInt(1), monitorIdentity, member, "REJECT", "Expired");
```

Comment:
not sure if I follow this logic. we're processing all the entries however, if any one of the entries fails, we mark the result as failed and don't execute the delete command. However, we have already inserted all those log entries claiming that the requests were rejected due to being expired. so the correct options would be:
a) either wrap this in a transaction where in case of any errors from insertRoleAuditLog we return right away and in DBService property rollback the changes (the problem with this method is that would we ever get into a case where we can never cleanup because one entry keeps failing?).
b) remove the logic to remove all entries at once and instead return the list to DBService and there we can have one delete operation (just like delete member) per entry wrapped in transaction.